packages/ifpack2/src/Ifpack2_BlockTriDiContainer_def.hpp

```diff
@@ -1278,9 +1278,16 @@ class Impl {
       if (collective_) {
         std::copy(wrk_.begin(), wrk_.begin() + n, wrk_.begin() + n);
 #ifdef HAVE_MPI
+# if MPI_VERSION >= 3
         MPI_Iallreduce(wrk_.data() + n, wrk_.data(), n,
                        Teuchos::Details::MpiTypeTraits<magnitude_type>::getType(),
                        MPI_SUM, comm_, &mpi_request_);
+# else
+# pragma message "MPI_Iallreduce is not available."
```

Comment:
Please use `Tpetra::idot` (in `tpetra/core/src/Tpetra_idot.hpp`) or `Tpetra::Details::iallreduce` (in  `tpetra/core/src/Tpetra_Details_iallreduce.hpp`).  It has a fall-back for the `MPI_VERSION < 3` case, which we still need to support.  This would avoid the need for checking whether MPI exists.